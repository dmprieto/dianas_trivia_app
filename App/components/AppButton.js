import { Text, TouchableOpacity } from "react-native"

export const AppButton = ({ onPress, title }) => (
  <TouchableOpacity
    onPress={onPress}
    className="bg-purple-800 hover:bg-purple-500 py-2 px-4 rounded opacity-80"
  >
    <Text className="text-xl text-white font-bold">{title}</Text>
  </TouchableOpacity>
)
