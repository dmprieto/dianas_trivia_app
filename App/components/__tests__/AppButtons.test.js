import renderer from "react-test-renderer"

import { AppButton } from "../AppButton"

describe("<AppButton />", () => {
  it("has 1 child", () => {
    const tree = renderer.create(<AppButton />).toJSON()
    expect(tree.children.length).toBe(1)
  })

  it("renders correctly", () => {
    const tree = renderer.create(<AppButton />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
