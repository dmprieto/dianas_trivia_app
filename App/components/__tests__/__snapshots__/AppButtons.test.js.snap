// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppButton /> renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#6b21a8",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "opacity": 0.8,
      "paddingBottom": 8,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 8,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 20,
          "lineHeight": 28,
        },
        Object {
          "color": "#fff",
        },
        Object {
          "fontWeight": "700",
        },
      ]
    }
  />
</View>
`;
